lib/ruport/data/record.rb

```diff
@@ -55,10 +55,10 @@ def initialize(data,options={})
       case(data)
       when Array
         @attributes = options[:attributes] || (0...data.length).to_a
-        @data = @attributes.inject({}) { |h,a| h.merge(a => data.shift) }
+        @data = @attributes.each_with_object({}) { |a, h| h[a.freeze] = data.shift }
       when Hash
-        @data = data.dup
-        @attributes = options[:attributes] || data.keys
+        @data = data
```

Comment:
Do we know what happens if the data gets modified outside of here?

Do we use this record and then pretty much throw it away?